integration/kotlinx-coroutines-guava/src/ListenableFuture.kt

```diff
@@ -77,88 +292,143 @@ private class ListenableFutureCoroutine<T>(
 }
 
 /**
- * Converts this deferred value to the instance of [ListenableFuture].
- * The deferred value is cancelled when the resulting future is cancelled or otherwise completed.
+ * A [ListenableFuture] that delegates to an internal [DeferredListenableFuture], collaborating with
+ * it.
+ *
+ * This setup allows the returned [ListenableFuture] to maintain the following properties:
+ *
+ * - Correct implementation of [Future]'s happens-after semantics documented for [get], [isDone]
+ *   and [isCancelled] methods
+ * - Cancellation propagation both to and from [Deferred]
+ * - Correct cancellation and completion semantics even when this [ListenableFuture] is combined
+ *   with different concrete implementations of [ListenableFuture]
+ *   - Fully correct cancellation and listener happens-after obeying [Future] and
+ *     [ListenableFuture]'s documented and implicit contracts is surprisingly difficult to achieve.
+ *     The best way to be correct, especially given the fun corner cases from
+ *     [AsyncFuture.setAsync], is to just use an [AsyncFuture].
```

Comment:
There is no such class as `AsyncFuture`

Reply:
AbstractFuture. Not sure how I got that one wrong.

Fixed.